### PR TITLE
Add CITATION.cff

### LINKED ISSUE #5518: Citation of Triton inference server
**Is your feature request related to a problem? Please describe.**
Could you add a citation link to your Github repo? (.cff file or bibtex) I want to cite it for our research publication and do not know who to put in the authors. I checked the issues but could not find anything relevant. 

**Describe the solution you'd like**
Github has a feature of adding a cff file. I think that would be the easiest. 

**Describe alternatives you've considered**
This is a bibtex I came up with. 
@misc{triton,
  author = {John Doe?},
  title = {Triton Inference Server},
  year = {2023},
  publisher = {NVIDIA},
  version= {x},
  journal = {GitHub repository},
  howpublished = {\url{https://github.com/triton-inference-server/server}},
  commit = {6aa17924c7ef47f4e24053459fcabbc501f646f2}
}

### PR DESCRIPTION
**Pull request by rmccorm4**
Fixes #5518 

Followed [NeMo's example](https://github.com/NVIDIA/nemo/blob/main/CITATION.cff).

Used `NVIDIA Corporation` as generic author.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,35 @@
+cff-version: 1.2.0
+message: "If you use this software, please cite it as below."
+title: "Triton Inference Server: an optimized cloud and edge inferencing solution."
```
Comment (dzier) [inline CITATION.cff:3]: The subtext needs to be in proper title format. "An Optimized Cloud and Edge Inference Solution"

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+url: https://github.com/triton-inference-server
+repository-code: https://github.com/triton-inference-server/server
+authors:
+  - family-names: Goodwin
```
Comment (dzier) [inline CITATION.cff:7]: For authors, we just need to add "NVIDIA Corporation". We shouldn't call out individual engineers. If this was a specific paper, that would be different, however, there are a lot of contributors. We should just be generic here.

**Review by dzier — CHANGES_REQUESTED** [review]

**Comment by rmccorm4** [comment]
Updated @dzier 

Generated citations:

APA:
```
NVIDIA Corporation. Triton Inference Server: An Optimized Cloud and Edge Inferencing Solution. [Computer software]. https://github.com/triton-inference-server/server
```
Bibtex:
```
@software{NVIDIA_Corporation_Triton_Inference_Server,
author = {{NVIDIA Corporation}},
title = {{Triton Inference Server: An Optimized Cloud and Edge Inferencing Solution.}},
url = {https://github.com/triton-inference-server/server}
}
```

**Review by dyastremsky — APPROVED** [review]

**Review by dzier — APPROVED** [review]

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,7 @@
+cff-version: 1.2.0
+message: "If you use this software, please cite it as below."
+title: "Triton Inference Server: An Optimized Cloud and Edge Inferencing Solution."
+url: https://github.com/triton-inference-server
+repository-code: https://github.com/triton-inference-server/server
+authors:
+  - name: "NVIDIA Corporation"
```
Comment (jbkyang-nvi) [inline CITATION.cff:7]: ` NVIDIA CORPORATION & AFFILIATES` ?

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,7 @@
+cff-version: 1.2.0
+message: "If you use this software, please cite it as below."
+title: "Triton Inference Server: An Optimized Cloud and Edge Inferencing Solution."
```
Comment (jbkyang-nvi) [inline CITATION.cff:3]: Where does this line come from?

Comment (rmccorm4) [reply CITATION.cff:3]: This is our official "About" description on the github repo

image

File: CITATION.cff (hunk 3 above)
Comment (rmccorm4) [reply CITATION.cff:7]: @dzier should we add `NVIDIA Corporation & Affiliates` ?

Comment (rmccorm4) [reply CITATION.cff:7]: resolved in sprint meeting

**Pull request merged**